packages/lab/src/dropdown-next/DropdownNext.tsx

```diff
@@ -1,254 +1,418 @@
+      setFocusedState(true);
+
+      onFocus?.(event);
+    };
+
+    const ignoreBlur = useRef(false);
+
+    const handleBlur = (event: FocusEvent<HTMLButtonElement>) => {
+      if (!ignoreBlur.current) {
+        setOpen(event, false);
+      }
+      ignoreBlur.current = false;
+
+      setFocusedState(false);
+      onBlur?.(event);
+    };
+
+    const handleListMouseOver = () => {
+      setFocusVisibleState(false);
+    };
+
+    const handleListMouseDown = () => {
+      ignoreBlur.current = true;
+    };
+
+    const handleListFocus = () => {
+      buttonRef.current?.focus();
+    };
+    const handleListClick = () => {
+      buttonRef.current?.focus();
+    };
+
+    useEffect(() => {
+      if (openState && !activeState) {
+        if (selectedState.length > 0) {
+          const selected = getOptionsMatching(
+            (option) => option.value === selectedState[0]
+          ).pop();
+          if (selected) {
+            setActive(selected);
+          }
+        } else {
+          setActive(getOptionAtIndex(0));
+        }
+      } else if (!openState) {
+        setActive(undefined);
+      }
+    }, [openState, children]);
+
+    const listId = useId();
+    const handleListRef = useForkRef<HTMLDivElement>(floating, listRef);
+
+    return (
+      <div
+        className={clsx(withBaseName(), withBaseName(variant), {
+          [withBaseName("disabled")]: disabled,
+          [withBaseName(validationStatus ?? "")]: validationStatus,
         })}
-        value={selectedItem}
-        className={clsx(
-          withBaseName("button"),
-          withBaseName(variant),
-          {
-            [withBaseName("disabled")]: disabled,
-            [withBaseName("readOnly")]: readOnly,
-          },
-          className
-        )}
-        role="combobox"
-        aria-haspopup="listbox"
-        aria-expanded={open}
-        aria-activedescendant={activeDescendant}
-        tabIndex={disabled ? -1 : 0}
-        aria-owns={listId}
-        aria-controls={listId}
-        aria-disabled={disabled}
-        {...restProps}
-        ref={triggerRef}
       >
-        <span className={clsx(withBaseName("buttonText"))}>{selectedItem}</span>
-        {getIcon()}
-      </button>
-      <FloatingComponent
-        open={open && !disabled}
-        ref={portalRef}
-        {...getDropdownNextProps()}
-        {...getPosition()}
-      >
-        <ListNext
-          data-test-id={"list-container"}
-          id={listId}
-          className={clsx(withBaseName("list"), ListProps?.className)}
-          disableFocus
-          disabled={disabled || ListProps?.disabled}
-          selected={selectedItem}
-          highlightedItem={highlightedItem}
-          {...ListProps}
-          ref={setListRef}
-          onSelect={handleListSelect}
-        >
-          {getListItems(source)}
-        </ListNext>
-      </FloatingComponent>
-    </div>
-  );
-});
+        <ListControlContext.Provider value={listControl}>
+          <button
```

Comment:
Add `type="button"` otherwise clicking Dropdown within a form will cause the form to submit